File: workspace/src/controllers/UserController.js
import { ref } from 'vue';

export default function useUserController() {
  const users = ref([]);
  const loading = ref(false);
  const error = ref(null);

  const fetchUsers = async () => {
    loading.value = true;
    error.value = null;
    try {
      const res = await fetch('https://jsonplaceholder.typicode.com/users');
      if (!res.ok) throw new Error('Error al cargar usuarios');
      users.value = await res.json();
    } catch (err) {
      error.value = err.message;
    } finally {
      loading.value = false;
    }
  };

  return { users, loading, error, fetchUsers };
}

File: workspace/backend/user_controller.py
import requests

class UserController:
    def __init__(self):
        self.users = []
        self.loading = False
        self.error = None

    def fetch_users(self):
        self.loading = True
        self.error = None
        try:
            res = requests.get('https://jsonplaceholder.typicode.com/users')
            res.raise_for_status()
            self.users = res.json()
        except Exception as err:
            self.error = str(err)
        finally:
            self.loading = False
        return self.users
